dtfs-central-api/src/v1/controllers/tfm/facility/tfm-update-facility.controller.js

```diff
@@ -10,20 +15,30 @@ const withoutId = (obj) => {
   return cleanedObject;
 };
 
-const updateFacility = async (facilityId, tfmUpdate) => {
+const updateFacility = async ({ facilityId, tfmUpdate, sessionPortalUser, sessionTfmUser, isSystemUpdate }) => {
   const collection = await db.getCollection(DB_COLLECTIONS.TFM_FACILITIES);
 
+  let auditDetails = {};
+  if (isSystemUpdate) {
+    auditDetails = generateSystemAuditDetails();
```

Comment:
If we changed how we create the object in the tfm-api, we could handle the creation of this object with a single function that then has a switch case based on `userType` -- making this a bit cleaner? What do you think?

```
const auditDetails = getAuditDetailsFromAuditInformation(auditInformation)
```